metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala

```diff
@@ -289,61 +309,163 @@ object SemanticTokensProvider {
-    var delta = Line(0, 0)
+  def makeScalaCliTokens(
+      comm: Token.Comment,
+      initialDelta: Line,
+  ): (List[Integer], Line) = {
+    val cliTokens = getTokens(false, comm.value)
     val buffer = ListBuffer.empty[Integer]
-    val cliTokens = tokens.takeWhile(_.isWhiteSpaceOrComment)
-    val rest = tokens.drop(cliTokens.length)
-    def makeInitialTokens(tk: Token) = {
-      tk match {
-        case comm: Token.Comment if comm.value.startsWith(">") =>
-          val cliTokens = getTokens(false, comm.value)
-          cliTokens.foreach { tk =>
-            val (toAdd, delta0) = tk match {
-              case start: Token.Ident if start.value == ">" =>
-                convertTokensToIntList(
-                  "//>",
-                  delta,
-                  getTypeId(SemanticTokenTypes.Comment),
-                )
-              case using: Token.Ident if using.value == "using" =>
-                convertTokensToIntList(
-                  using.value,
-                  delta,
-                  getTypeId(SemanticTokenTypes.Keyword),
-                )
-              case tk =>
-                val (tpe, mod) = typeModOfNonIdentToken(tk, false)
-                convertTokensToIntList(
-                  tk.text,
-                  delta,
-                  tpe,
-                  mod,
-                )
-            }
-            buffer.addAll(
-              toAdd
-            )
-            delta = delta0
-          }
-        case _ =>
+    var delta = initialDelta
+    cliTokens.foreach { tk =>
+      val (toAdd, delta0) = tk match {
+        case start: Token.Ident if start.value == ">" =>
+          convertTokensToIntList(
+            "//>",
+            delta,
+            getTypeId(SemanticTokenTypes.Comment),
+          )
+        case using: Token.Ident if using.value == "using" =>
+          convertTokensToIntList(
+            using.value,
+            delta,
+            getTypeId(SemanticTokenTypes.Keyword),
+          )
+        case tk =>
           val (tpe, mod) = typeModOfNonIdentToken(tk, false)
-          val (toAdd, delta0) = convertTokensToIntList(
+          convertTokensToIntList(
             tk.text,
             delta,
             tpe,
             mod,
           )
-          buffer.addAll(
-            toAdd
-          )
-          delta = delta0
       }
+      buffer.addAll(
+        toAdd
+      )
+      delta = delta0
     }
-    cliTokens.foreach(makeInitialTokens)
-    (delta, buffer.toList, rest)
+    (buffer.toList, delta)
   }
 
+  def makeDocStringTokens(
+      comm: Token.Comment,
+      initialDelta: Line,
+  ): (List[Integer], Line) = {
+    val docstring = "/*" + comm.value + "*/"
+    val buffer = ListBuffer.empty[Integer]
+    val paramOrThrows: String = "(@param|@tparam|@throws)\\s+([\\w.]+)"
+    val apiLink: String = "\\[\\[(.*)\\]\\]"
+    val other: String = "(@[a-zA-Z]+)"
+    val reg: Regex = s"$paramOrThrows|$apiLink|$other".r
```

Comment:
You could construct the regex in an object, so it would not be created every time